src/AutoRest.CSharp/Mgmt/Generation/MgmtExtensionWrapperWriter.cs

```diff
@@ -1,36 +1,105 @@
+        private MgmtExtensionWrapper This { get; }
+        public MgmtExtensionWrapperWriter(MgmtExtensionWrapper extensionsWrapper) : base(new CodeWriter(), extensionsWrapper)
         {
-            _wrapper = extensionsWrapper;
+            This = extensionsWrapper;
+        }
+
+        protected override void WritePrivateHelpers()
+        {
+            var generalExtensionParameter = new Parameter(
+                "resource",
+                null,
+                typeof(ArmResource),
+                null,
+                ValidationType.None,
+                null);
+            var armClientParameter = new Parameter(
+                "client",
+                null,
+                typeof(ArmClient),
+                null,
+                ValidationType.None,
+                null);
+            var scopeParameter = new Parameter(
+                "scope",
+                null,
+                typeof(ResourceIdentifier),
+                null,
+                ValidationType.None,
+                null);
+            foreach (var extensionClient in This.ExtensionClients)
+            {
+                if (extensionClient.IsEmpty)
+                    continue;
+
+                _writer.Line();
+
+                var extensionClientSignature = new MethodSignature(
+                    $"Get{extensionClient.Type.Name}",
+                    null,
+                    null,
+                    Private | Static,
+                    extensionClient.Type,
+                    null,
+                    new[] { generalExtensionParameter });
+                using (_writer.WriteMethodDeclaration(extensionClientSignature))
+                {
+                    using (_writer.Scope($"return {generalExtensionParameter.Name}.GetCachedClient(client =>", newLine: false))
+                    {
+                        _writer.Line($"return new {extensionClientSignature.ReturnType}(client, {generalExtensionParameter.Name}.Id);");
+                    }
+                    _writer.LineRaw(");");
+                }
+
+                _writer.Line();
+
+                var scopeExtensionClientSignature = new MethodSignature(
+                    $"Get{extensionClient.Type.Name}",
+                    null,
+                    null,
+                    Private | Static,
+                    extensionClient.Type,
+                    null,
+                    new[] { armClientParameter, scopeParameter });
+                using (_writer.WriteMethodDeclaration(scopeExtensionClientSignature))
+                {
+                    using (_writer.Scope($"return {armClientParameter.Name}.GetResourceClient(() => ", newLine: false))
+                    {
+                        _writer.Line($"return new {extensionClientSignature.ReturnType}({armClientParameter.Name}, {scopeParameter.Name});");
+                    }
+                    _writer.LineRaw(");");
+                }
```

Comment:
To be refactored - move these method signatures into the MgmtClientExtension class